### ISSUE TITLE
bootstrap automatic user creation doesn't work with Ubuntu

### ISSUE THREAD

**Issue by jbicha** [debops/ansible-bootstrap#43]
On a default Ubuntu cloud image, the only account that can log in is named 'ubuntu'. Root does not have a password and even if it had a password can't be used to log in via ssh. The 'ubuntu' user must use sudo (but without the need to type in the user passwords) to run admin

It works to use debops bootstrap -u ubuntu -b but bootstrap then assumes that the 'ubuntu' user is what I want to use from then on, instead of adding my current username as an admin user. I really would like to use my current username instead of the placeholder 'ubuntu'.

Suggestions
---------------
1. Have bootstrap on a Ubuntu remote host try the ubuntu username first. Then proceed to add the current username as an admin user by default.

2. Maybe debops bootstrap shouldn't need any other options by default. No "-u root" on Debian oldstable, no "-u ubuntu -b" on Ubuntu.

3. Update the docs to explain how to get this to work on Ubuntu, or improve how it works and update the docs to match the improved behavior.

I think the [docs](https://docs.debops.org/en/latest/debops-playbooks/docs/guides/getting-started.html) are incomplete right now, because they say that bootstrap doesn't work on Debian stable without pre-configuration but they don't explain what specifically needs to be changed either. Now that Debian has official cloud images, I think bootstrap should be set to automatically work with them or give specific instructions on how to pre-configure them.

**Comment by drybjed** [comment]
The `debops.bootstrap` role and playbook are meant to be used to prepare a host for Ansible/DebOps management. Usually you would use it for the first host in a new environment which then can be configured as PXE/preseed server which other host can use to bootstrap themselves. Therefore the playbook doesn't assume anything about the remote host and lets you specify what user account to use, optional password needed for configuration, use of `sudo`, etc. Different environments have different configuration so it's hard to select a good default for this.

The `debops.bootstrap` role can now manage multiple admin accounts at once. What you might want to do is add your own admin account through the Ansible inventory. When the role will bootstrap a host, it will manage the `ubuntu` account you specify on the command line, as well as create your own admin account which you then should be able to use to manage the host.

The issue with Debian requiring a manual modification of the SSH configuration to let you connect over `root` account with password is a hard bootstrapping problem. Let's say that you install a new Debian host without an user account, only with `root` password-protected. How do you connect to that host over SSH when by default Debian Installer disallows password access for `root` account? There's nothing Ansible alone can help you with. You either need to fix your installation process (`cloud-init` configuration, PXE/preseeding, custom Debian Installer image, etc.) or allow access manually.

I'm not sure about Debian cloud images either, since the DebOps environment is used on normal, physical hosts as well. This issue is entirely dependent on your bootstrap/installation process and should be fixed at this stage.

The documentation about this definitely could be improved, I'll look into that in the future.

**Comment by jbicha** [comment]
I'm using cloud-init now instead of debops.bootstrap since I'm basically using Ubuntu cloud images and the different providers I've been using (EC2, lxd, etc.) support it.

**Comment by drybjed** [comment]
Great, so does `cloud-init` let you pass an SSH public key to the specified user accounts? You can use that to make unatteneded connections work out of the box.

Perhaps there should a generic `cloud-init` DebOps repository with a script that lets you generate your own setup as needed. I'll look into it when `cloud-init` is introduced in the DebOps environment.

**Comment by jbicha** [comment]
Yes. here's a minimal [cloud-init](https://cloudinit.readthedocs.io/en/latest/) file that can be used for bootstrapping.

```
#cloud-config
users:
  - name: jeremy
    gecos: Jeremy Bicha
    groups: sudo,adm,admin
    shell: /bin/bash
    sudo: ['ALL=(ALL) NOPASSWD:ALL']
    ssh-import-id: github:jbicha
disable_root: true
fqdn: webserver.example.com
package_upgrade: true
package_reboot_if_required: true
packages:
  - python
```

**Comment by drybjed** [comment]
Excellent, that will be very handy. Thank you! :-)
